index.js

```diff
@@ -187,6 +187,9 @@ function use (extensions) {
     this.on('close', onfinalize)
     this.on('end', onfinalize)
     this.on('finish', this.finalize)
+    this.on('error', function (err) {
+      debug('stream error', err)
+    })
```

Comment:
There must have been some handler swallowing these errors somewhere, because until I added the debug() there was no exception in the process (just silent hypercore-protocol failures).

If we check if debug is enabled first, I think that's a good compromise, because it means the executor is watching for the event somewhere. It means DEBUG=* or DEBUG=hypercore-protocol env vars are set.